app/controllers/data_controller.rb

```diff
@@ -12,38 +12,60 @@ def choose_data
   end
 
   def display_data
-    info = []
-    params["title"].each do |title|
-      url = "http://api.worldbank.org/countries/#{params[:country]}/indicators/#{title}?per_page=500&date=1960:2016&format=json"
-      info << JSON.parse(Net::HTTP.get_response(URI(url)).body)[1]
-    end
+    if params["title"].nil?
+      flash["notice"] = "You didn't choose any datasets/countries?!"
+      redirect_to data_path
+    elsif params["title"].count > 1 && params["title"].count <= 3 && params["country"].count == 1
+      info = []
+      params["title"].each do |title|
+        url = "http://api.worldbank.org/countries/#{params[:country][0]}/indicators/#{title}?per_page=500&date=1960:2016&format=json"
+        info << JSON.parse(Net::HTTP.get_response(URI(url)).body)[1]
+      end
 
-    @infos_data = []
-    info.each do |dataset|
-      set = {}
-      data = []
-      dataset.map do |datum|
-        data << [DateTime.parse("#{datum['date']}-01-01 00:00:00").to_i*1000, datum["value"].to_f] if datum["value"]
+      @infos_data = []
+      info.each do |dataset|
+        set = {}
+        data = []
+        dataset.map do |datum|
+          data << [DateTime.parse("#{datum['date']}-01-01 00:00:00").to_i * 1000, datum["value"].to_f] if datum["value"]
+        end
+        set["title"] = dataset[1]["indicator"]["value"]
+        set["country"] = dataset[1]["country"]["value"]
+        set["data"] = data
+        @infos_data << set
+      end
+    elsif params["title"].count == 1 && params["country"].count > 1
+      info = []
+      @no_info = []
+      params["country"].each do |country|
+        url = "http://api.worldbank.org/countries/#{country}/indicators/#{params["title"][0]}?per_page=500&date=1960:2016&format=json"
```

Comment:
Line is too long. [134/80]<br>Prefer single-quoted strings inside interpolations.